File: patterns/Python/pattern8.py
"""
Generate the following pattern
*
* *
* * *
* * * *
* * * * *
* * * *
* * *
* *
*
"""

def pattern8(n):
    for i in range(1, n+1):
        for j in range(i):
            print("* ",end="")
        print("")
    
    for i in range(n+1, 2*n):
        for j in range(2*n-i):
            print("* ",end="")
        print("")
    
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern8(n)
    
if __name__ == "__main__":
    main()
           

File: patterns/C++/pattern8.cpp
/*
Generate the following pattern
    *
    * *
    * * *
    * * * *
    * * * * *
    * * * *
    * * *
    * *
    *
*/

#include<bits/stdc++.h>
using namespace std;

void pattern8(int n){
    for(int i=1; i<=n; i++){
        for(int j=0; j<i; j++){
            cout << "* ";
        }
        cout << endl;
    }

    for(int i=n+1; i<2*n; i++){
        for(int j=0; j<2*n-i; j++){
            cout << "* ";
        }
        cout << endl;
    }

}

int main(){
    int n;
    cout << "Enter the Number of rows: ";
    cin >> n;

    cout << "\nHere is the generated pattern\n";
    pattern8(n);
}